### DISCUSSION TITLE
Difference between Coupled and CoupledValue

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#22010] [Q&A General] [answered]
Hello,

Just a minor question here,
_T(coupled("temperature"))
_T(coupledValue("temperature"))

are they same? I tried to find "coupled" this keyword in moose but could not.

Then another thing is the following "jvar" defined in the bracket, does it has a initial value there?
if not, how to exactly understand the meaning of  "if (jvar == _T)".

Real
Newmatrix :: computeQpOffDiagJacobian(unsigned int jvar)
{
  Real j = 0;

  if (jvar == _T)
  {
    j += _drho_dT[_qp] ;
    j *= _test[_i][_qp];
  }

  return j;
}

**Comment by GiudGiud** [comment, marked as answer]
Hello

_coupled gets you the index of the variable in the system, which is why it s appropriate to compare to jvar

coupledValue gets you the value at local quadrature points

Guillaume

**Reply by GiudGiud** [reply]
jvar does not have an initial value. It's the routine's argument. You should have a look at the doxygen to understand what the arguments to the routines mean
https://mooseframework.inl.gov/docs/doxygen/moose/classes.html

When you build the equation's Jacobian, you have to keep track of variable indices, especially on the off-diagonal. This is what _T is doing for temperature
